packages/vite/src/node/ssr/fetchModule.ts

```diff
@@ -139,11 +144,8 @@ function inlineSourceMap(
     return result
 
   // to reduce the payload size, we only inline vite node source map, because it's also the only one we use
-  const OTHER_SOURCE_MAP_REGEXP = new RegExp(
-    `//# ${SOURCEMAPPING_URL}=data:application/json[^,]+base64,([A-Za-z0-9+/=]+)$`,
-    'gm',
-  )
-  while (OTHER_SOURCE_MAP_REGEXP.test(code))
+  OTHER_SOURCE_MAP_REGEXP.lastIndex = 0
+  if (OTHER_SOURCE_MAP_REGEXP.test(code))
     code = code.replace(OTHER_SOURCE_MAP_REGEXP, '')
```

Comment:
IIUC `OTHER_SOURCE_MAP_REGEXP.test(code)` will only be `true` once, so this `while` can simply be `if`.

Reply:
It can be several times by mistake :P